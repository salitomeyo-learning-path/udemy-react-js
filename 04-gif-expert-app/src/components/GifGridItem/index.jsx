import PropTypes from 'prop-types';

// import './styles.css';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className='card'>
            <img className='gif' src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
