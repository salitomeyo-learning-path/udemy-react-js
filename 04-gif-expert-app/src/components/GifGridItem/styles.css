.gif {
    width: 200px;
    height: 200px;
    border-radius: 10px;
}

.card {
    background-color: #d0d0d0;
    border-radius: 10px;
    margin: 10px;
    padding-bottom: 10px;
    width: 200px;
}
  
.card p {
    background-color: #d0d0d0;
    color: #3f3f3f;
    text-align: center;
    margin: 5px;
}