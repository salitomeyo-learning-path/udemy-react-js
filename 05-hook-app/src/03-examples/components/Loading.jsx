
export const Loading = () => {
    return(
        <div className="alert alert-info text-center">
            Loading...
            <div class="spinner-border text-secondary" role="status">
                <span class="visually-hidden">Loading...</span>
            </div>
        </div>
    )
}

