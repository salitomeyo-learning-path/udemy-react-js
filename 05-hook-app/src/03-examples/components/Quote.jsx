
export const Quote = ({ quote, author }) => {
    return (
        <blockquote className="blockquote text-end">
            <p className="m-1">{ quote }</p>
            <footer className="blockquote-footer mt-1">{ author }</footer>
        </blockquote>
    )
}

