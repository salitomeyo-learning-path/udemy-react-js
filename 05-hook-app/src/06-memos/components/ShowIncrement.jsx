import { memo } from "react";

export const ShowIncrement = memo(({ increment }) => {
    console.log('me volvi a generar :c');

    return(
        <button
            className="btn btn-primary"
            onClick={() => increment( 5 )}
        >
            Increment
        </button>
    )
})

