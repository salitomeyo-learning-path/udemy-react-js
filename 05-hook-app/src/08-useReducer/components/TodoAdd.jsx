import { useState } from "react"
import { useForm } from "../../hooks/useForm";

export const TodoAdd = ({ onNewTodo }) => {

    const { formState, onInputChange, onResetForm } = useForm({ description: ' ' })
    const { description } = formState;

    const onSubmitForm = ( event ) => {
        event.preventDefault();

        if ( description.length <= 1) return;

        const newTodo = {
            id: new Date().getTime(),
            description,
            done: false
        }

        onNewTodo && onNewTodo( newTodo );
        onResetForm();
    }

    return (
        <form onSubmit={onSubmitForm}>
            <input 
                type="text"
                placeholder='Que hay que hacer?'
                className='form-control'
                name='description'
                value={description}
                onChange={ onInputChange }
            />
            <button 
                type="submit"
                className='btn btn-outline-primary my-2'
            >
                Agregar
            </button>
        </form>
    )
}

