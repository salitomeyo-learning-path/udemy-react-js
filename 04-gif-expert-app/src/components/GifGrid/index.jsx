import PropTypes from 'prop-types';
import { useFetchGifs } from "../../hooks/useFetchGifs";
import { GifGridItem } from "../GifGridItem";

// import './styles.css';

export const GifGrid = ({ category }) => {

    const { gifs, isLoading } = useFetchGifs( category );
    
    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading ?
                    <p>Cargando...</p>
                :
                    <div className="card-grid">
                    { 
                        gifs.map( (gif) => (
                            <GifGridItem 
                                key={gif.id}  
                                { ...gif }
                            />
                        ))
                    }
                    </div>

            }
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
