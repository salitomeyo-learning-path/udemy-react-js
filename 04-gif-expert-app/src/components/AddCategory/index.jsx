import { useState } from "react"

import './styles.css';

export const AddCategory = ({ onNewCategory }) => {
    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    const onSubmit = ( event ) => {
        event.preventDefault();
        if( inputValue.trim().length <= 1 ) return;
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return(
        <form className="search-container" onSubmit={ (event) => onSubmit(event) }>
            <input 
                type="text"
                placeholder="Search gifs" 
                value={ inputValue }   
                onChange = { (event) => onInputChange(event) }
                />
            <button type="submit">Agregar</button>
        </form>
    )
}
