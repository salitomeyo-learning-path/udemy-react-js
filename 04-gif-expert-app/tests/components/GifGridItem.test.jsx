import { render, screen } from '@testing-library/react';
import { GifGridItem } from "../../src/components/GifGridItem"

describe('Testing <GifGridItem />', () => {

    const title = 'Deku';
    const url = 'https://static.wikia.nocookie.net/boku-no-hero-academia-fanon/images/9/92/Deku.png/revision/latest?cb=20200728090057&path-prefix=es'

    test('Should match snapshot', () => {
        const { container } = render( <GifGridItem title={ title } url={ url }  />)
        expect( container ).toMatchSnapshot();
    })

    test('Should show image with indicated url', () => {
        render( <GifGridItem title={ title } url={ url }  /> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('Should show title in component', () => {
        render( <GifGridItem title={ title } url={ url }  /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    })
})
