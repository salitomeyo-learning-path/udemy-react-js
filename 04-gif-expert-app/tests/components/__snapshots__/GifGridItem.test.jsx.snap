// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem /> Should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Deku"
      class="gif"
      src="https://static.wikia.nocookie.net/boku-no-hero-academia-fanon/images/9/92/Deku.png/revision/latest?cb=20200728090057&path-prefix=es"
    />
    <p>
      Deku
    </p>
  </div>
</div>
`;
